src/quartz_api/cmd/main.py

```diff
@@ -189,20 +191,52 @@ def redoc_html() -> FileResponse:
     # Customize the OpenAPI schema
     server.openapi = lambda: _custom_openapi(server)
 
+    # Store auth instance for middleware
+    auth_instance = None
+
     # Override dependencies according to configuration
     match (conf.get_string("auth0.domain"), conf.get_string("auth0.audience")):
         case (_, "") | ("", _):
-            server.dependency_overrides[auth.get_auth] = auth.DummyAuth()
+            auth_instance = auth.DummyAuth()
+            server.dependency_overrides[auth.get_auth] = auth_instance
             log.warning("disabled authentication. NOT recommended for production")
         case (domain, audience):
-            server.dependency_overrides[auth.get_auth] = auth.Auth0(
+            auth_instance = auth.Auth0(
                 domain=domain,
                 api_audience=audience,
                 algorithm="RS256",
             )
+            server.dependency_overrides[auth.get_auth] = auth_instance
         case _:
             raise ValueError("Invalid Auth0 configuration")
 
+
+# middleware to add user id and email to sentry
+    @server.middleware("http")
+    async def add_user_details_to_sentry(
+        request: Request,
+        call_next: Callable[[Request], Awaitable[Response]],
+    ) -> Response:
+        if auth_instance is not None and not isinstance(auth_instance, auth.DummyAuth):
+            try:
+                authorization = request.headers.get("Authorization", "")
+                if authorization.startswith("Bearer "):
+                    token = authorization.replace("Bearer ", "")
+                    credentials = HTTPAuthorizationCredentials(scheme="Bearer", credentials=token)
+                    payload = auth_instance(request, credentials)
+                    if payload:
+                        import sentry_sdk
+                        sentry_sdk.set_user({
+                            "id": payload.get("sub"),
+                            "email": payload.get(auth.EMAIL_KEY),
```

Comment:
have you been able to test this? 

To get a bearer token you can sign up to app.quartz.solar and then inspect one of the api routes to and see your bearer token, or use the example code [here](https://www.notion.so/openclimatefix/API-Access-2d8d2f64215d4432be830cbcc9220012)